sklearn/metrics/_dist_metrics.pyx.tp

```diff
@@ -562,9 +562,9 @@ cdef class DistanceMetric{{name_suffix}}(DistanceMetric):
 
                 D[i1, i2] = self.dist_csr(
                     x1_data,
-                    x1_indices,
+                    &x1_indices[0],
```

Comment:
Can't comment on the relevant line, but my thinking is that the argument to the function should be defined as

```
const int32_t[::1] x2_indices
```

Reply:
Yes, the arrays that we pass here and also in other routines must be contiguous.